src/main/java/com/cmlteam/serv/HttpHandlerFilesList.java

```diff
@@ -0,0 +1,121 @@
+package com.cmlteam.serv;
+
+import com.sun.net.httpserver.HttpExchange;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class HttpHandlerFilesList extends HttpHandlerBase {
+
+    private final File[] files;
+    private Deque<String> referers = new ArrayDeque<>();
+
+    private String[] HEADER = {"<h1>Files List</h1>",
+            "<br></br>",
+            "<a href='/listing?name=back'>Go to upper level</a>",
+            "<br></br>",
+            "<table>",
+                "<thead>",
+                    "<tr>",
+                        "<th>File or Folder Name</th>",
+                        "<th>Size</th>",
+                    "</tr>",
+                "</thead>",
+            "<tbody>"};
+    private final String[] FOOTER = {"</tbody>",
+                                "</table>"
+    };
+
+    public HttpHandlerFilesList(File[] files) {
+        this.files = files;
+    }
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+        log(httpExchange);
+        httpExchange.getResponseHeaders().add("Content-Type", "text/html; charset=utf-8");
+        httpExchange.sendResponseHeaders(200, 0);
+        OutputStream os = httpExchange.getResponseBody();
+        String param = findRequestParam(httpExchange.getRequestURI().toString());
+        File nextFolder;
+        switch (param) {
+            case "toplevel":
+                showList(files, os);
+                break;
+            case "back":
+                if (referers.size() < 2) {
+                    referers.clear();
+                    showList(files, os);
+                } else {
+                    referers.pop();
+                    nextFolder = new File (referers.peek());
+                    showList(nextFolder.listFiles(), os);
+                }
+                break;
+            default:
+                referers.push(param);
+                nextFolder = new File(param);
```

Comment:
There is no validity check to provided path. It means the user can list any folder on the system just by hand-crafting the url like `/listing?name=/etc/` or `/listing?name=/../../../../../../../../`. The user should only be able to access any folder **inside** the list of served folders.